app/controllers/peer_reviews_controller.rb

```diff
@@ -1,67 +1,147 @@
 class PeerReviewsController < ApplicationController
+  include GroupsHelper
+
   before_action :set_peer_review, only: [:show, :edit, :update, :destroy]
   before_filter :authorize_only_for_admin
 
-  # GET /peer_reviews
-  # GET /peer_reviews.json
   def index
-    @peer_reviews = PeerReview.all
+    @assignment = Assignment.find(params[:assignment_id])
+    @section_column = ''
+    if Section.all.size > 0
+      @section_column = "{
+        id: 'section',
+        content: '" + t(:'summaries_index.section') + "',
+        sortable: true
+      },"
+    end
   end
 
-  # GET /peer_reviews/1
-  # GET /peer_reviews/1.json
-  def show
+  def populate
+    @assignment = Assignment.find(params[:assignment_id])
+
+    reviewer_groups = get_groupings_table_info()
+    reviewee_groups = get_groupings_table_info(@assignment.parent_assignment)
+
+    reviewee_to_reviewers_map = create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    id_to_group_names_map = create_map_group_id_to_name(reviewer_groups, reviewee_groups)
+    num_reviews_map = create_map_number_of_reviews_for_reviewer(reviewer_groups)
+
+    render json: [reviewer_groups, reviewee_groups, reviewee_to_reviewers_map,
+                  id_to_group_names_map, num_reviews_map]
   end
 
-  # GET /peer_reviews/new
-  def new
-    @peer_review = PeerReview.new
+  # Returns a dict of: reviewee_id => [list of reviewers].
+  def create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    reviewer_ids = reviewer_groups.map { |reviewer| reviewer['id'] }
+    peer_review_map = Hash.new { |hash, key| hash[key] = [] }
+    reviewee_groups.each { |reviewee| peer_review_map[reviewee['id']] }
```

Comment:
The above may not be exactly what you wanted. Making the .map() work is probably possible, though I felt I was spending too much time trying to learn how to make something work that already works.
If you know how to do it however let me know, I'd like to learn it and use it.